src/Backend/Core/Language/LanguageName.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Backend\Core\Language;
+
+use InvalidArgumentException;
+
+final class LanguageName
```

Comment:
I am not yet convinced by this. I would prefer to just call this Language but in that case the name of the Language class should change, and I wasn't sure to what. Maybe Locale or Translations ?